packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1087,15 +1093,21 @@ export class SnapController extends BaseController<
         preinstalled: true,
       });
 
-      // Setup permissions
-      const processedPermissions = processSnapPermissions(
+      // Process Initial and Dynamic permissions
+      const processedInitialPermissions = processSnapPermissions(
         manifest.initialPermissions,
       );
-
-      this.#validateSnapPermissions(processedPermissions);
+      this.#validateSnapPermissions(processedInitialPermissions);
```

Comment:
We should validate dynamic permissions too

Reply:
Added validation step now. But, I had to modify the validation function.